src/request.js

```diff
@@ -136,4 +136,26 @@ module.exports = class Request extends EventEmitter {
     }
     return null;
   }
+
+  // Temporarily suspends the flow of data from the database.
+  // No more 'row' events will be emitted until resume() is called.
+  pause() {
+    if (!this.isActive()) {
+      return;
+    }
+    this.connection.pauseTokenStreamParser();
+  }
+
+  // Resumes the flow of data from the database.
+  resume() {
+    if (!this.isActive()) {
+      return;
+    }
+    this.connection.resumeTokenStreamParser();
+  }
+
+  // Returns true if this request is the currently active request of the connection.
+  isActive() {
+    return this.connection && this.connection.request === this && this.connection.state === this.connection.STATE.SENT_CLIENT_REQUEST;
+  }
 };
```

Comment:
This introduces pretty deep coupling between the Connection class and Request class. Perhaps this suggests pause/resume methods should be on the Connection class?